### PR TITLE
#159203295 Revisit the search asset endpoints

### PR DESCRIPTION
**Pull request by purpose50**
#### What does this PR do?
Improve the search asset endpoint

#### Description of Task to be completed?
* create an immutableMultiDict to mock request.args
* clean up the dynamic filter by removing the logic for search asset
* modify the search asset endpoint to improve the code for readability

#### How should this be manually tested?
* pull from this branch ```bg-revisit-the-search-asset-endpoints-159203295``` into a new branch
* run ``` pip install -r requrements.txt ```
* endpoint should look like this: ```/assets/search?start=<date>&end=<date>&warranty_start=<date>&warranty_end=<date>```
*date format should be ```Y-M-D```
* run ```pytest ``` to run the test

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[159203295](https://www.pivotaltracker.com/story/show/159203295)

#### Postman collection:
- [Link](https://www.getpostman.com/collections/2f53fc56d99f1ae2f2c8)

#### Screenshots (if appropriate)
screen shot 2018-07-11 at 7 53 31 am

screen shot 2018-07-23 at 11 43 48 am

#### Questions:
N/A

### CODE REVIEW

File: api/views/asset.py [hunk 1]
```diff
@@ -55,13 +55,39 @@ def get(self):
...
+                raise ValidationError(
+                    dict(message=filter_errors['INVALID_COLUMN'].format(key)))
+
+        start = request.args.get('start')
+        end = request.args.get('end')
+        warranty_start = request.args.get('warranty_start')
+        warranty_end = request.args.get('warranty_end')
+
+        qry_list = []
+        if start:
```
Comment (SEUNAGBEYE) [inline api/views/asset.py]: @purpose50 Could you please give this a try:

```
qry_list = []

queries = {
  'start': request.args.get('start'),
  'end': request.args.get('end'),
  'warranty_start': request.args.get('warranty_start'),
  'warranty_end': request.args.get('warranty_end')
}


def query_object(query):
  where = 'where'
  created_at = 'created_at'
  queries = {
    'start': (where, f'{created_at},ge,{query}'),
    'end': (where, f'{created_at},le,{query}'),
    'warranty_start': (where, '{created_at},ge,{query}'),
    'warranty_end': (where, '{created_at},le,{query}')
  }

  return queries[query]

  for query, value in queries:
    if value:
      qry_list.append(query_object(query))
```
With the code above we could eliminate the long `if(s)`

Comment (purpose50) [reply api/views/asset.py]: Nice one @SEUNAGBEYE , but I think the existing code is simpler and easier to read than the suggested one.

Comment (SEUNAGBEYE) [reply api/views/asset.py]: @purpose50 Yes, that is true.

**Comment by brayoh** [comment]
@purpose50 fix the merge conflicts and also update the description with the new postman collection link

**Pull request merged**